src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -754,15 +807,20 @@ ucs_status_t uct_tcp_sockcm_ep_create(const uct_ep_params_t *params, uct_ep_h *e
         tcp_sockcm = uct_tcp_sockcm_ep_get_cm(tcp_ep);
         UCS_ASYNC_BLOCK(tcp_sockcm->super.iface.worker->async);
 
-        /* the server's endpoint was already created by the listener, return it */
-        *ep_p  = &tcp_ep->super.super.super;
-
         /* fill the tcp_ep fields from the caller's params */
-        status = uct_cm_set_common_data(&tcp_ep->super, params);
+        status = uct_cm_use_ep_params(&tcp_ep->super, params);
+        if (status != UCS_OK) {
+            goto err;
+        }
+
+        status = uct_tcp_sockcm_ep_server_update_cm(tcp_ep, params->cm);
```

Comment:
can we do this outside UCS_ASYNC_UNBLOCK?
since during this function, the cep does not belong to same async context anymore

Reply:
might be a bit risky to first set the UCT_TCP_SOCKCM_EP_SERVER_CREATED flag and then change the cm.
what if an error event arrives on the old cm (on the async thread) before the end of this function - it will handle it wrong 
https://github.com/openucx/ucx/blob/master/src/uct/tcp/tcp_sockcm.c#L76

Reply:
can we remove from old cm, then set the flag, then add to new cm?

Reply:
only in case of different cms yes? 
the regular flow shouldn't remove from old cm, set the flag and set the old one again.

Reply:
yes, if same cm should just set the flag

Reply:
is it possible that after removing the from old handler and setting this flag, an event will arrive to the old handler from the missed queue? (if it got there during the uct_ep_create function, while the lock was taken by the main thread)

Reply:
not sure, pls check it

Reply:
@yosefe looks like it cant:
when an fd is removed from the handlers, which tcp_sockcm would do before setting the UCT_TCP_SOCKCM_EP_SERVER_CREATED  flag, it's removed from its khash from this call - https://github.com/openucx/ucx/blob/master/src/ucs/async/async.c#L536 .
so then, if the missed queue would try to invoke a handler, it would first have to get its fd from the khash, but it wouldn't be there already - https://github.com/openucx/ucx/blob/master/src/ucs/async/async.c#L622